openedx/core/djangoapps/credit/api/eligibility.py

```diff
@@ -225,6 +228,15 @@ def set_credit_requirement_status(username, course_key, req_namespace, req_name,
             )
 
     """
+    username = user.username
```

Comment:
Why do we need this line? Why can't we just use `user.username` for all places in this function where username is needed?